websites/T/TruckersMP/presence.ts

```diff
@@ -1,380 +1,149 @@
-      item = document.querySelector(
-        "body > div.container > div > div > ol > li.breadcrumb-item.active"
-      );
-      item2 = document.querySelector("#stats-players");
-      output = document.querySelector("#stats-time");
-      presenceData.details = "Traffic Stats, viewing server:";
-      presenceData.state =
-        item.innerText + " (Online: " + item2.innerText + ")";
-      delete presenceData.smallImageKey;
-
-      presence.setActivity(presenceData);
-    } else {
-      presenceData.details = "Viewing Traffic Stats";
-      delete presenceData.state;
-      delete presenceData.smallImageKey;
-
-      presence.setActivity(presenceData);
+    //EVENTS CENTER
+    else if (document.location.pathname == "/events") {
+      presenceData.details = "Browsing the events center";
+    } else if (document.location.pathname == "/events/manage") {
+      delete presenceData.smallImageKey;
+      presenceData.smallImageKey = "editing";
+      presenceData.details = "Managing their events";
+    } else if (document.location.pathname == "/events/manage/past") {
+      presenceData.details = "Viewing their past events";
+    } else if (document.location.pathname.includes("/events/create")) {
+      delete presenceData.smallImageKey;
+      presenceData.smallImageKey = "editing";
+      presenceData.details = "Creating an event";
+    } else if (document.location.pathname.includes("/events/search")) {
+      delete presenceData.smallImageKey;
+      presenceData.smallImageKey = "search";
+      presenceData.details = "Searching for an event";
+    } else if (document.location.pathname.includes("/events/")) {
+      presenceData.details = "Viewing an event";
+      presenceData.state = document.querySelectorAll("h1")[1].innerText;
+      presenceData.buttons = [{ label: "Event", url: document.URL }];
+    }
+  } else if (document.location.hostname == "forum.truckersmp.com") {
+    if (document.location.pathname.includes("/forum/")) {
+      presenceData.details = "Browsing a forum category";
+      presenceData.state = document.querySelector("header > h1").innerHTML;
+      presenceData.buttons = [{ label: "Forum Category", url: document.URL }];
+    } else if (document.location.pathname.includes("/topic/")) {
+      presenceData.details = "Viewing a forum topic";
+      presenceData.state = document.querySelector(
+        "main > div > div > div > div > div > div > h1 > span.ipsType_break > span"
+      ).innerHTML;
+      presenceData.buttons = [{ label: "Forum Topic", url: document.URL }];
+    } else if (document.location.pathname.includes("/profile/")) {
+      presenceData.details = "Viewing a forum user";
+      presenceData.state = document
+        .querySelectorAll(
+          "main > div > div > div> div > header > div  > div > div"
+        )[1]
+        .querySelector("div > h1 > span").innerHTML;
+      presenceData.buttons = [{ label: "Forum User", url: document.URL }];
```

Comment:
What even is this forum user looks like grammar error to me, but make it something else

Reply:
Nah, you have website users and you have forum users too

Reply:
kekw so forum user is the one who uses forum